theme/static/js/resourceLandingAjax.js

```diff
@@ -1591,23 +1593,86 @@ function setFileTypeSpatialCoverageFormFields(logical_type){
+            url = url.replace('update-file-metadata', 'delete-file-coverage');
+            $btnDeleteSpatialCoverage.unbind('click');
+            $btnDeleteSpatialCoverage.click(function () {
+                deleteFileTypeSpatialCoverage(url, $btnDeleteSpatialCoverage);
+
+            })
+        };
+
+        var addSpatialCoverageLink = function () {
+            var $spatialForm = $("#id-coverage-spatial-filetype");
+            var deleteLink = '<a id="id-btn-delete-spatial-filetype" type="button" style="display: block;" class="pull-right"><span class="glyphicon glyphicon-trash icon-button btn-remove"></span>';
+            $spatialForm.find('legend').html('Spatial Coverage' + deleteLink);
+            $btnDeleteSpatialCoverage = $("#id-btn-delete-spatial-filetype");
+            return $btnDeleteSpatialCoverage;
+        };
+        // set spatial form attribute 'data-coordinates-type' to point or rectangle
+        if ($id_type_filetype_div.find(radioBoxSelector).attr("checked") === "checked"){
+            $("#id-coverage-spatial-filetype").attr('data-coordinates-type', 'rectangle');
+            // check if spatial coverage exists
+            var $btnDeleteSpatialCoverage = $("#id-btn-delete-spatial-filetype");
+            if(!$btnDeleteSpatialCoverage.length) {
+                // delete option doesn't exist
+                $btnDeleteSpatialCoverage = addSpatialCoverageLink();
+            }
+            var formSpatialCoverage = $btnDeleteSpatialCoverage.closest('form');
+            var url = formSpatialCoverage.attr('action');
+            if(url.indexOf('update-file-metadata') !== -1) {
+                onSpatialCoverageDelete();
+            }
+            else {
+                $btnDeleteSpatialCoverage.hide()
+            }
+        }
+        else {
+            $("#id-coverage-spatial-filetype").attr('data-coordinates-type', 'point');
+            $btnDeleteSpatialCoverage = $("#id-btn-delete-spatial-filetype");
+            if(!$btnDeleteSpatialCoverage.length) {
+                $btnDeleteSpatialCoverage = addSpatialCoverageLink();
+            }
+            formSpatialCoverage = $btnDeleteSpatialCoverage.closest('form');
+            url = formSpatialCoverage.attr('action');
+            if(url.indexOf('update-file-metadata') !== -1) {
+                onSpatialCoverageDelete();
+            }
+            else {
+                $btnDeleteSpatialCoverage.hide()
+            }
+        }
+        if(bindCoordinatesPicker){
+            $("#id-coverage-spatial-filetype").coordinatesPicker();
+        }
     }
 
     // #id_type_1 is the box radio button
-    if ($id_type_filetype_div.find("#id_type_1").attr("checked") == "checked" ||
-        (logical_type != 'GeoFeatureLogicalFile' && logical_type != 'RefTimeseriesLogicalFile' && logical_type != 'GenericLogicalFile')) {
-        // coverage type is box
+    if ($id_type_filetype_div.find(radioBoxSelector).attr("checked") === "checked" ||
```

Comment:
This only change to the if statement we need is "#id_type_1" -> radioBoxSelector rest of the if statement should not be changed.